File: DekafMain/src/sql/SqlCommand.java
package org.jetbrains.dekaf.sql;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.function.Function;



/**
 * A SQL command.
 *
 * @author Leonid Bushuev from JetBrains
 */
public class SqlCommand extends SqlStatement {

  public SqlCommand(@NotNull final TextFragment sourceFragment) {
    super(sourceFragment);
  }

  public SqlCommand(@NotNull final String sourceText, final int row, @Nullable final String commandName) {
    super(sourceText, row, commandName);
  }


  public SqlCommand(@NotNull final String sourceText) {
    super(sourceText);
  }


  private SqlCommand(final int row,
                     @NotNull final String sourceText,
                     @Nullable final String name,
                     @NotNull final String description) {
    super(row, sourceText, name, description);
  }


  @NotNull
  @Override
  public SqlCommand rewrite(@NotNull final Function<String,String> operator) {
    String transformedSourceTex = operator.apply(mySourceText);
    return new SqlCommand(myRow, transformedSourceTex, myName, myDescription);
  }

}

File: dekaf-main/src/sql/SqlCommand.kt
package org.jetbrains.dekaf.sql

import org.jetbrains.dekaf.text.TextFragment

import java.util.function.Function


/**
 * A SQL command.
 *
 * @author Leonid Bushuev
 */
class SqlCommand : SqlStatement {


    constructor(sourceFragment: TextFragment) : super(sourceFragment)


    constructor(sourceText: String, row: Int, commandName: String?) : super(sourceText, row, commandName)


    constructor(sourceText: String) : super(sourceText)


    private constructor(row: Int,
                        sourceText: String,
                        name: String?,
                        description: String)
            : super(row, sourceText, name, description)


    override fun rewrite(operator: Function<String, String>): SqlCommand {
        val transformedSourceTex = operator.apply(sourceText)
        return SqlCommand(row, transformedSourceTex, name, description)
    }

}
